decidim-core/spec/content_parsers/decidim/user_parser_spec.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+module Decidim
+  describe ContentParsers::UserParser do
+    let(:user) { create(:user, :confirmed) }
+    let(:parser) { described_class.new(content) }
+
+    context "when mentioning a valid user" do
+      let(:content) { "This text contains a valid user mention: @#{user.nickname}" }
+
+      it "rewrites the mention" do
+        expect(parser.rewrite).to include(user.to_global_id.to_s)
```

Comment:
Can you check for the exact string to ensure the rest of the string is not modified?

`expect(...).to eq "This text contains a valid user mention: @#{user.to_global_id.to_s}"`